airbyte-cdk/python/airbyte_cdk/sources/lcc/configurable_stream.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.lcc.retrievers.retriever import Retriever
+from airbyte_cdk.sources.lcc.schema.schema_loader import SchemaLoader
+from airbyte_cdk.sources.streams.core import IncrementalMixin, Stream
+
+
+class ConfigurableStream(Stream, IncrementalMixin):
+    """
+    ConfigurableStream is a Stream that delegates most of its logic to its schema_load and retriever
+    """
+
+    def __init__(self, name, primary_key, cursor_field, schema_loader: SchemaLoader, retriever):
+        self._name = name
+        self._primary_key = primary_key
+        self._cursor_field = cursor_field
+        self._schema_loader = schema_loader
+        self._retriever: Retriever = retriever
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self._primary_key
+
+    @property
+    def name(self) -> str:
+        """
+        :return: Stream name. By default this is the implementing class name, but it can be overridden as needed.
+        """
+        return self._name
+
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return self._retriever.get_state()
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        """This method is only needed to interface with AbstractSource..."""
+        pass
```

Comment:
@sherifnada 
I left it as `pass` so the retriever is the only one that's responsible for updating the state (https://github.com/airbytehq/airbyte/pull/12850/files#diff-7e502fbdea4963678f0a0a3fa2c7ad5af221259b89a375a06db2e35dd5d5d750R212-R219)

would it make more sense for the retriever not to update the state and let the abstract source take care of it?

Reply:
I just meant that the `IncrementalMixin` interface which this implements expects the setter to actually store the state, so I was a bit surprised to see we're `pass`ing here. I think it's fine if the retriever updates that state for DRY reasons as long as this method delegates to its impl . LMK if it would be helpful to sync on `IncrementalMixin` IRL -- it basically makes the stream's state an instance var in the stream aka "stateful" in the programming sense

Reply:
Got it. this doesn't need the mixin since the retriever is updating the state.